bin/check_spdx.py

```diff
@@ -62,11 +62,18 @@ def main(src):
         sys.exit(1)
 
 
-def get_files_in_dir(path: Path, root: Path, gitignore: PathSpec) -> List[Path]:
-    assert root.is_absolute(), f"INTERNAL ERROR: `root` must be absolute but is {root}"
+def get_gitignore_files():
+    command = ["git", "ls-files", "--others", "--ignored", "--exclude-standard"]
+    result = subprocess.run(command, capture_output=True, text=True, check=True)
+    return [line for line in result.stdout.strip().split("\n") if line]
+
+
+def get_files_in_dir(path: Path, root: Path, gitignore_files: List[str]) -> List[Path]:
```

Comment:
The return type is incorrect I think, it should be a Generator
![afbeelding](https://github.com/user-attachments/assets/0b03abbe-f633-4b83-9bc1-a2c04d507c51)